libdebug/cffi/ptrace_cffi_source.c

```diff
@@ -49,6 +71,74 @@ struct global_state {
     _Bool handle_syscall_enabled;
 };
 
+struct thread *get_thread(struct global_state *state, int tid)
+{
+    struct thread *t = state->t_HEAD;
+    while (t != NULL) {
+        if (t->tid == tid) return t;
+        t = t->next;
+    }
+
+    return NULL;
+}
+
+struct fp_regs_struct *get_thread_fp_regs(struct global_state *state, int tid)
+{
+    struct thread *t = get_thread(state, tid);
+
+    if (t) {
+        return &t->fpregs;
+    }
+
+    return NULL;
+}
+
+void get_fp_regs(struct global_state *state, int tid)
+{
+    struct thread *t = get_thread(state, tid);
```

Comment:
If you just need `t->fpregs`, why iterate through all the threads each time instead of passing the reference obtained with `get_thread_fp_regs` called by the Python?